### PR TITLE
perf(codegen): slightly faster `print_list`

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

File: crates/oxc_codegen/src/lib.rs [hunk 1]
```diff
@@ -412,11 +412,13 @@ impl<'a> Codegen<'a> {
...
     fn print_list<T: Gen>(&mut self, items: &[T], ctx: Context) {
-        for (index, item) in items.iter().enumerate() {
-            if index != 0 {
-                self.print_comma();
-                self.print_soft_space();
-            }
+        let mut iter = items.iter();
+        if let Some(item) = iter.next() {
+            item.gen(self, ctx);
+        }
```
Comment (overlookmotel) [inline crates/oxc_codegen/src/lib.rs:418]: No need to fall through to the `while` loop if `iter` is empty.

```suggestion
        let Some(item) = iter.next() else { return; };
        item.gen(self, ctx);
```

**Review by DonIsaac — APPROVED** [review]

**Comment by DonIsaac** [comment]
I'm seeing a minor performance degradation in our benchmarks. Could this be causing `rustc` to hit a de-opt path?
image

**Comment by Boshen** [comment]
> I'm seeing a minor performance degradation in our benchmarks. Could this be causing `rustc` to hit a de-opt path? image

This is just noise.

**Comment by overlookmotel** [comment]
I don't think it is just noise. The benchmark time went up to 21 ms on this PR and then sat there steady across the following ~10 commits. [CodSpeed](https://codspeed.io/oxc-project/oxc/benchmarks/tasks/benchmark/benches/codegen.rs::codegen::bench_codegen::codegen%5Bchecker.ts%5D?branchId=66cb07dd8f2e0b608676753e)

From the flamegraph it looks like making this function more complex has caused it not to get inlined any more.

Maybe adding `#[inline]` would fix that.

**Pull request merged**